### ISSUE TITLE
FlurryAgent: Listener cannot be null

### ISSUE THREAD

**Issue by vidicunt** [flurry/flurry-adapter-admob-android#2]
Hello!

I am trying to use Flurry to mediate banners, however I keep getting these errors (in order, selected which relevant)

FlurryAgent: Listener cannot be null
FlurryAgent: New main file also not found. returning..
a: onError(com.flurry.android.ads.FlurryAdBanner@629b0f0FETCH20)
FlurryAdapter: Calling onFailedToReceiveAd for Banner with errorCode: 3
FlurryAdapter: Destroy Ad

Thanks!

EDIT: Interstitials work fine

**Comment by KimiChiu** [comment]
@tommyzat Did you resolve this problem?

**Comment by vidicunt** [comment]
Nope, I stopped using Flurry.

**Comment by KimiChiu** [comment]
That's too bad.
I found that it just need to add these initialization codes.

    new FlurryAgent.Builder()
                .withLogEnabled(false)
                .withCaptureUncaughtExceptions(true)
                .withContinueSessionMillis(10000)
                .withListener(new FlurryAgentListener() {
                    @Override
                    public void onSessionStarted() {
                        Timber.d("FlurryAgent session started.");
                    }
                })
                .build(this, getResources().getString(R.string.flurry_api_key));

But I still get `Error code: 3` which means `No context - context is missing`.
Totally don't know what it means.
A year ago, I got `no fill` every time, same as two years ago.
It's just so difficult to integrate Flurry banner into AdMod.

**Comment by vidicunt** [comment]
Well, the solution is to just not use them. They're the ones losing money
until a proper API is set in place.

**Comment by KimiChiu** [comment]
I just tried to use native ad as a banner ad in my app.
And it works well with Flurry.
Maybe it's time to get rid of WebView ads.
Don't know if it has the same ecpm as before.

**Comment by tactoth** [comment]
@tommyzat you did the best thing, i've been mediating several ad networks in the past few years, and flurry gives way more frustration a fraction of other network (fb/admob/mopub's revenue). had faith in yahoo but so far proven they never cared. the sdk is always broken.

**Comment by vidicunt** [comment]
I'm using admob only for now. I used to mediate with Facebook, however
their policies were insane and couldn't conform.

I really want them to launch native ads already..

**Comment by KimiChiu** [comment]
I was only using admob too. But then it applied "fat-finger" features to my account.
After that, I almost can't pay my cloud services fees.
Even though I totally followed there instructions to display ads. 
Adding margin on banners, don't use sandwich layout design, etc.
That's why I quit AdMob banner.

I'm now using Facebook, Flurry, MoPub and AdMob natives at the same time.
But the AdMob has the lowest priority.
Not only because the fat-finger feature, but also they're having the worst layout performance.
Right now, Facebook and Flurry natives are working well.
I still got "ads no fill: error code 20" from Flurry natives for about 20 hours yesterday.
So I think the best way is not to count on only one mediation network.

**Comment by tactoth** [comment]
I've been using Admob, facebook, mopub and flurry for a while, all native & interstitial, Admob pays really well for interstitial, and facebook pays ok for native. and yes the problem with admob native ads is the layout is painfully slow (flurry is doing pretty well here), so i have to use a 0.75 multiplier on the cpm reported to show less admob native ads.

Flurry's major problem so far:
- many regressions (no test at all?), with 10.1.0 i was able to get ads, now when I upgrade to 11.1.1, i get many 'fetch' error 20, and ads are simply not filled (if it's a backend fill problem why 10.1.0 doesn't have this issue). This happens many times in the last few years.
- many request sent without response, started from ~80% request going w/o response, then I found there was one bug that the ad listener is probably weak referenced, made an fix, however i'm still seeing many such case from data logging (no clue so far), the issue existed for ~3 years.
- cpm down down down, while it makes sense to invest in a fragile sdk by try to figure out how the sdk makes  assumptions without any documenting, when it's daily revenue drops from peak at ~$300 to $10, it's not worthing any effort anymore.

**Comment by KimiChiu** [comment]
Same problem happens on AdMob.
There's is a "signal 5" bug which crashed my app thousands times a day since last year.
And they still can't fix it.
After they applied fat-finger feature on my account, the daily revenue never across $50.
Millions of impressions a day still can't get enough revenue to pay all the fees?
That's ridiculous.

**Comment by flurrydev** [comment]
(I do not know what fat-finger feature is and whether it relates to Flurry or not) 
Please be aware that as of May-25-2018, the  GDPR compliance has impacted the fill for all EU users. Refer to https://developer.yahoo.com/flurry/docs/publisher/gdpr/ for instructions provide user consent for the Flurry ad serving.  

@tactoth , @KimiChiu if you have Flurry related integration problem(s), please write to integration@flurry.com and provide your API key along with the summary of the problem.

**Comment by KimiChiu** [comment]
@flurrydev What's the consent string? Is there a complete example of setting GDPR?

**Comment by KimiChiu** [comment]
@tommyzat @tactoth You're right. Flurry is too unstable. The fill rate dropped to 0% in just one day. It's already stopped serving ads for 6 days. I didn't do anything since two months ago.
